model/DataStore/PersistDataService.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoDeliveryRdf\model\DataStore;
+
+use common_exception_Error;
+use common_exception_NotFound;
+use oat\oatbox\filesystem\FileSystem;
+use oat\oatbox\filesystem\FileSystemService;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\helpers\FileHelperService;
+use tao_helpers_Uri;
+use taoQtiTest_models_classes_export_TestExport22;
+
+class PersistDataService extends ConfigurableService
+{
+    private const DATA_STORE = 'dataStore';
+    private const DELIVERY_META_DATA_JSON = 'deliveryMetaData.json';
+    private const TEST_META_DATA_JSON = 'testMetaData.json';
+    private const ITEM_META_DATA_JSON = 'itemMetaData.json';
+    private const PACKAGE_FILENAME = 'QTIPackage';
+    private const ZIP_EXTENSION = '.zip';
+
+    public function persist(array $params): void
+    {
+        $fileSystem = $this->getDataStoreFilesystem();
+        $folder = $this->getFolderName($params['deliveryId']);
+
+        $this->persistData($fileSystem, $folder, self::DELIVERY_META_DATA_JSON, $params['deliveryMetaData']);
```

Comment:
We should keep this in try catch block.

Reply:
If we will try/catch here, we can remote the try/catch here too: 

https://github.com/oat-sa/extension-tao-delivery-rdf/pull/368/files#diff-d7dda33bcb2b0511001c7d9b6fa95a73ce058a954084c4bf1db51ae715a22fb0R84

I think try/catch should be done in the client class (using this one) not here. Or we will have repeated logs. We could remove logging from this class too.